featurebyte/service/feature_offline_store_info.py

```diff
@@ -0,0 +1,238 @@
+from featurebyte.query_graph.transform.decompose_point import FeatureJobSettingExtractor
+from featurebyte.query_graph.transform.offline_store_ingest import (
+    OfflineStoreIngestQueryGraphTransformer,
+)
+from featurebyte.service.catalog import CatalogService
+from featurebyte.service.entity_serving_names import EntityServingNamesService
+from featurebyte.service.offline_store_feature_table import OfflineStoreFeatureTableService
+from featurebyte.service.offline_store_feature_table_construction import (
+    OfflineStoreFeatureTableConstructionService,
+)
+
+
+class OfflineStoreInfoInitializationService:
+    """
+    OfflineStoreInfoInitializationService class
+    """
+
+    def __init__(
+        self,
+        catalog_service: CatalogService,
+        offline_store_feature_table_service: OfflineStoreFeatureTableService,
+        offline_store_feature_table_construction_service: OfflineStoreFeatureTableConstructionService,
+        entity_serving_names_service: EntityServingNamesService,
+    ) -> None:
+        self.catalog_service = catalog_service
+        self.offline_store_feature_table_service = offline_store_feature_table_service
+        self.offline_store_feature_table_construction_service = (
+            offline_store_feature_table_construction_service
+        )
+        self.entity_serving_names_service = entity_serving_names_service
+
+    async def offline_store_feature_table_creator(
+        self,
+        primary_entity_ids: Sequence[ObjectId],
+        feature_job_setting: Optional[FeatureJobSetting],
+        has_ttl: bool,
+        catalog_id: ObjectId,
+        entity_id_to_serving_name: Optional[Dict[ObjectId, str]] = None,
+    ) -> str:
+        """
+        Create offline store feature table name
+
+        Parameters
+        ----------
+        primary_entity_ids: Sequence[ObjectId]
+            Primary entity ids
+        feature_job_setting: Optional[FeatureJobSetting]
+            Feature job setting
+        has_ttl: bool
+            Whether the offline store feature table has ttl
+        catalog_id: ObjectId
+            Catalog id
+        entity_id_to_serving_name: Optional[Dict[ObjectId, str]]
+            Entity id to serving name mapping
+
+        Returns
+        -------
+        str
+        """
+        construct_service = self.offline_store_feature_table_construction_service
+        catalog = await self.catalog_service.get_document(catalog_id)
+        table = await construct_service.get_dummy_offline_store_feature_table_model(
+            primary_entity_ids=primary_entity_ids,
+            feature_job_setting=feature_job_setting,
+            has_ttl=has_ttl,
+            feature_store_id=catalog.default_feature_store_ids[0],
+            catalog_id=catalog_id,
+            entity_id_to_serving_name=entity_id_to_serving_name,
+        )
+        persist_table = await self.offline_store_feature_table_service.get_or_create_document(
+            data=table
+        )
```

Comment:
Does this mean offline store feature tables will be created here and no longer by `OfflineStoreFeatureTableManagerService`? I think this change has some impact, for example table comments are added by `OfflineStoreFeatureTableManagerService` when it creates the tables, but that is not done here.

Is it possible to determine the table name here but defer the creation of the tables to `OfflineStoreFeatureTableManagerService`?

Reply:
It will create a table without any `feature_ids`. To determine the table name, we need to add records to the persistent. If the table comment is the only concern, could we update the logic to add table comments? It seems to be easier to fix than deferring table creation.

Reply:
The only way that i can think of to defer creation is to use the table signature. Based on our previous discussion, it seems to have big impact on current code changes...

Reply:
Could i tag a new table here https://github.com/featurebyte/featurebyte/blob/d9438b829743fb83705484e77d016ab3c8be044e/featurebyte/service/offline_store_feature_table_manager.py#L201 if the `feature_ids` of the original table is empty?

Reply:
Updating the table comments logic makes sense to me. That is the only issue I can see now.

Going this route, I think we should also update `OfflineStoreFeatureTableManagerService` to remove the code here:

https://github.com/featurebyte/featurebyte/blob/401c941b14af0cdd3e7a8dcac46ee1707a69264e/featurebyte/service/offline_store_feature_table_manager.py#L208-L220

since it will never be executed.

Reply:
Do we need to handle the case where the feature created before this change? If yes, maybe we should keep the code here...

Reply:
With the current implementation I don't feel comfortable for `offline_store_feature_table_service.create_document` to be called anywhere except by `OfflineStoreInfoInitializationService`, since the created table's name can be completely different from the provided `feature_table_model`. I think that can lead to inconsistency and probably not worth supporting.

Reply:
As long as it is ok to drop the support for the older feature, i am ok to remove that part.